app/models/ext_management_system.rb

```diff
@@ -99,6 +102,22 @@ def hostname_format_valid?
     errors.add(:hostname, _("format is invalid."))
   end
 
+  # validation - Zone cannot be changed when enabled == false
+  def zone_unchanged_when_paused?
+    return if changed_attributes.include?('enabled')
+
+    if changed_attributes.include?('zone_id') && !enabled?
+      errors.add(:zone, N_("cannot be changed when provider paused"))
+    end
+  end
+
+  # validation - Zone has to be visible when enabled == true
+  def zone_visible_when_active?
```

Comment:
Similar issue here with the naming. Perhaps `validate_ems_disabled_on_zone_change?` or something like that (also this gives preference to reuse the enabled/disabled wording instead of introducing active/inactive)